### PR TITLE
Branding (more customisable templates)

### PR DESCRIPTION
**Pull request by tiago-loureiro**
Please look at https://github.com/wireapp/wire-server/commit/a6451921d157beabc342e90aed2b91ca6d3b9fab for the actual changes, the other commit is very noisy due to changes in the templates.

Motivation: our templates have currently too much wire specific branding which needs adjustments in different environments: this PR caters for that by using the latest templates from https://github.com/wireapp/wire-emails (see https://github.com/wireapp/wire-emails/commit/8615420ef6fe366472f07525d73ea4a791d2e954 and https://github.com/wireapp/wire-emails/commit/817b079a696d276ff29affa4830a041f29c54964)

Note the new variables now:
```
brand
brand_url
brand_label_url
brand_logo
brand_service
copyright
forgot
legal
misuse
support
```

These are common to _all_ templates and thus require a few changes across the board. If nobody sees a nicer/better, I will just replace all calls to `renderText` and `renderHtml` with the new `renderTextWithBranding` and `renderHtmlWithBranding`

Worth mentioning that I tried sending out a verification email which looks good.

I did some work with @neongreen yesterday to refactor the email/sms/call code here https://github.com/wireapp/wire-server/compare/emails-v_1_51_0 but I'd like to leave a larger changeset for an independent refactor.

### CODE REVIEW

**Comment by neongreen** [comment]
LGTM

**Comment by neongreen** [comment]
Say when you want this to be re-reviewed

**Comment by tiago-loureiro** [comment]
> Say when you want this to be re-reviewed

Ready to go from my end, just needs some dependencies fixed to get a 💚 on the integration tests

**Comment by tiago-loureiro** [comment]
> Ready to go from my end, just needs some dependencies fixed to get a 💚 on the integration tests

All green now.

File: services/brig/deb/opt/brig/templates/de/provider/email/activation.txt [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
 Überprüfe [${url}]Falls du nicht auf den Button klicken kannst, kopiere den Link
 und füge ihn in deinem Browser ein:
 
 ${url}
 
-Solltest du dich nicht für eine Wire Service-Provider-Konto mit dieser
-E-Mail-Adresse registriert haben, kannst du diese Meldung ignorieren. Falls du
-einen Missbrauch deiner E-Mail-Adresse melden möchtest kontaktiere uns bitte.
-[https://support.wire.com/hc/de/requests/new]
+Solltest du dich nicht für eine ${brand} Service-Provider-Konto mit dieser
```
Comment (neongreen) [inline services/brig/deb/opt/brig/templates/de/provider/email/activation.txt]: Seems weird that it doesn't match activation-subject.txt (`${brand_service}-Benutzerkonto`).

File: services/brig/deb/opt/brig/templates/de/team/email/invitation.txt [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
 nachstehenden Button um die Einladung zu akzeptieren.
 
 Einladung akzeptieren [${url}]Falls du nicht auf den Button klicken kannst,
 kopiere den Link und füge ihn in deinem Browser ein:
 
 ${url}
 
-Wenn Du Fragen hast, bitte kontaktiere uns
-[https://support.wire.com/hc/de/requests/new].
+If you have any questions, please contact us [${support}].
```
Comment (neongreen) [inline services/brig/deb/opt/brig/templates/de/team/email/invitation.txt]: English! 😱

File: services/brig/deb/opt/brig/templates/de/team/email/new-member-welcome.txt [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
-Wire herunterladen [${url}]Falls du nicht auf den Button klicken kannst, kopiere
-den Link und füge ihn in deinem Browser ein:
+${brand} herunterladen [${url}]Falls du nicht auf den Button klicken kannst,
+kopiere den Link und füge ihn in deinem Browser ein:
 
 ${url}
 
-Wenn Du Fragen hast, bitte kontaktiere uns
-[https://support.wire.com/hc/de/requests/new].
+If you have any questions, please contact us [${support}].
```
Comment (neongreen) [inline services/brig/deb/opt/brig/templates/de/team/email/new-member-welcome.txt]: @giphy armageddon

File: services/brig/deb/opt/brig/templates/en/team/email/new-member-welcome.txt [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
+
+${brand_label_url} [${brand_url}]
+
 WELCOME TO ${team_name}.
-You have just joined a team called ${team_name} on Wire with ${email}.
+You have just joined a team called ${team_name} on ${brand} with ${email}.
 
-Wire combines strong encryption, a rich feature set and ease-of-use in one app
-like never before. Works on all popular platforms.
+${brand} combines strong encryption, a rich feature set and ease-of-use in one
```
Comment (neongreen) [inline services/brig/deb/opt/brig/templates/en/team/email/new-member-welcome.txt]: he-he-he

File: services/brig/src/Brig/Template.hs [hunk 5]
```diff
@@ -108,11 +116,23 @@ readText f = catchJust (\e -> if isDoesNotExistError e then Just () else Nothing
                        (readFile f)
                        (\_ -> error $ "Missing file: '" ++ f)
 
+-- | Uses a replace and a branding function, to replaces all placeholders from the
```
Comment (neongreen) [inline services/brig/src/Brig/Template.hs]: > to replaces

File: services/brig/src/Brig/Template.hs [hunk 6]
```diff
@@ -108,11 +116,23 @@ readText f = catchJust (\e -> if isDoesNotExistError e then Just () else Nothing
...
+-- given template to produce a Text. To be used on plain text templates
+renderTextWithBranding :: Template -> (Text -> Text) -> TemplateBranding -> Lazy.Text
```
Comment (neongreen) [inline services/brig/src/Brig/Template.hs]: Btw we have `LText` and `LByteString` imported in all modules by default now.

File: services/brig/src/Brig/Template.hs [hunk 7]
```diff
@@ -33,6 +39,8 @@ import qualified Data.Text.Lazy     as Lazy
 import qualified Data.Text.Template as Template
 import qualified HTMLEntities.Text  as HTML
 
+type TemplateBranding = Text -> Text
```
Comment (neongreen) [inline services/brig/src/Brig/Template.hs]: Maybe just `-- | See 'genTemplateBranding'.` would be enough.

File: services/brig/deb/opt/brig/templates/de/team/email/invitation.txt (hunk 2 above)
Comment (tiago-loureiro) [reply services/brig/deb/opt/brig/templates/de/team/email/invitation.txt]: Thank u!

**Comment by tiago-loureiro** [comment]
Blocked until the templates get fixed

File: services/brig/brig.integration.yaml [hunk 8]
```diff
@@ -73,7 +73,18 @@ emailSMS:
     templateDir: deb/opt/brig/templates
     emailSender: backend-integration@wire.com
     smsSender: "<insert-sender-number-here>"
-
+    templateBranding:
+      brand: Wire
+      brandUrl: https://wire.com
+      brandLabel: wire.com
```
Comment (neongreen) [inline services/brig/brig.integration.yaml]: This should go away

Comment (tiago-loureiro) [reply services/brig/brig.integration.yaml]: Thanks!

**Pull request merged**